### PR TITLE
[BIG PR] Gradio blocks & redesigned components

### PR DESCRIPTION
**Pull request by abidlabs**
Opening this PR as a draft so that we can start thinking about merging it into `main`. 

Of course, before we can merge it in, we need to:

- [x] Rewrite the core `Interface` class in terms of Blocks
- [x] Restore the full functionality of the `Interface` class 
- [x] Ensure that all old python unit tests pass (for backwards compatibility)
- [x] Redesign all of the components (at least at a basic level -- a full redesign can come later)

### CODE REVIEW

**Comment by FarukOzderim** [comment]
WoW!
image

**Comment by pngwn** [comment]
@abidlabs @FarukOzderim @aliabid94 

When we merge this in, do you think we could keep the commits intact (i.e. don't squash)? So it is easier to track commits -> PRs. I don't really want to root through the body of the commit message when looking through the history.

**Comment by abidlabs** [comment]
Yeah I don't think we typically squash our merges so it should be fine

**Comment by FarukOzderim** [comment]
@aliabid94 is there anything we need to review on this PR?

File: gradio/inputs.py [hunk 1]
```diff
@@ -488,3 +489,27 @@ def __init__(
             DeprecationWarning,
         )
         super().__init__(default_value=default, label=label)
+
+
+class Image3D(C_Model3D):
```
Comment (FarukOzderim) [inline gradio/inputs.py:494]: Do we need to add `Image3D` here? These are just for backwards compatibility.

File: test/test_inputs.py [hunk 2]
```diff
@@ -741,7 +741,7 @@ def test_in_interface(self):
 
 class TestImage3D(unittest.TestCase):
     def test_as_component(self):
-        Image3D = gr.test_data.BASE64_IMAGE3D
+        Image3D = media_data.BASE64_MODEL3D
```
Comment (FarukOzderim) [inline test/test_inputs.py:744]: it is better to `deepcopy` `test_data` in case any changes occur to them.

File: test/test_inputs.py [hunk 3]
```diff
@@ -777,7 +777,7 @@ def test_as_component(self):
             Image3D_input.serialize(Image3D, True)
 
     def test_in_interface(self):
-        Image3D = gr.test_data.BASE64_IMAGE3D
+        Image3D = media_data.BASE64_MODEL3D
```
Comment (FarukOzderim) [inline test/test_inputs.py:782]: it is better to `deepcopy` `test_data` in case any changes occur to them.

File: test/test_networking.py [hunk 4]
```diff
@@ -67,51 +67,51 @@ def test_start_server(self):
...
-        client = TestClient(app)
-        aiohttp.ClientSession.post = mock.MagicMock()
-        aiohttp.ClientSession.post.__aenter__ = None
-        aiohttp.ClientSession.post.__aexit__ = None
-        io.interpret = mock.MagicMock(return_value=(None, None))
-        response = client.post("/api/interpret/", json={"data": ["test test"]})
-        aiohttp.ClientSession.post.assert_called()
-        self.assertEqual(response.status_code, 200)
-        io.close()
+# class TestFlagging(unittest.TestCase):
```
Comment (FarukOzderim) [inline test/test_networking.py:70]: hmm, why did we comment out these tests?

File: test/test_outputs.py [hunk 5]
```diff
@@ -555,23 +555,14 @@ def test_as_component(self):
         )
         with tempfile.TemporaryDirectory() as tmpdirname:
             to_save = Image3D_output.save_flagged(
-                tmpdirname, "Image3D_output", gr.test_data.BASE64_IMAGE3D, None
+                tmpdirname, "Image3D_output", media_data.BASE64_MODEL3D, None
```
Comment (FarukOzderim) [inline test/test_outputs.py:558]: it is better to `deepcopy` `test_data` in case any changes occur to them.

File: test/test_outputs.py [hunk 6]
```diff
@@ -555,23 +555,14 @@ def test_as_component(self):
...
             )
             self.assertEqual("Image3D_output/0.gltf", to_save)
             to_save = Image3D_output.save_flagged(
-                tmpdirname, "Image3D_output", gr.test_data.BASE64_IMAGE3D, None
+                tmpdirname, "Image3D_output", media_data.BASE64_MODEL3D, None
```
Comment (FarukOzderim) [inline test/test_outputs.py:562]: it is better to `deepcopy` `test_data` in case any changes occur to them.

**Comment by FarukOzderim** [comment]
@aliabid94 this [test](
https://app.circleci.com/pipelines/github/gradio-app/gradio/2551/workflows/d6302915-e29c-4bec-a27e-129eac3da71d/jobs/2591?invite=true#step-115-35) started fail in another PR after your merge, did you had any changes about this function or data? If not it might be because of non deepcopied test_data I commented about above.

**Comment by abidlabs** [comment]
> @aliabid94 this [test](https://app.circleci.com/pipelines/github/gradio-app/gradio/2551/workflows/d6302915-e29c-4bec-a27e-129eac3da71d/jobs/2591?invite=true#step-115-35) started fail in another PR after your merge, did you had any changes about this function or data? If not it might be because of non deepcopied test_data I commented about above.

It was because certain files were reorganized in `blocks-dev` -- fixed in #1023

**Pull request merged**